src/Temporalio/Worker/ITemporalWorkerPlugin.cs

```diff
@@ -0,0 +1,48 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Temporalio.Worker
+{
+    /// <summary>
+    /// Interface for temporal worker plugins.
+    /// </summary>
+    /// <remarks>
+    /// WARNING: This API is experimental and may change in the future.
+    /// </remarks>
+    public interface ITemporalWorkerPlugin
+    {
+        /// <summary>
+        /// Gets the plugin name.
+        /// </summary>
+        public string Name { get; }
+
+        /// <summary>
+        /// Configures the worker options.
+        /// </summary>
+        /// <param name="options">The worker options to configure.</param>
+        public void ConfigureWorker(TemporalWorkerOptions options);
+
+        /// <summary>
+        /// Runs the worker asynchronously.
+        /// </summary>
+        /// <param name="worker">The worker to run.</param>
+        /// <param name="continuation">The continuation function.</param>
+        /// <returns>A task representing the asynchronous operation.</returns>
+        public Task RunWorkerAsync(TemporalWorker worker, Func<TemporalWorker, Task> continuation);
```

Comment:
```suggestion
        public Task<TResult> RunWorkerAsync<TResult>(TemporalWorker worker, Func<TemporalWorker, Task<TResult>> continuation, CancellationToken stoppingToken);
```

Arguably this is the full superset of the execute async call, even if our internals don't model it that way (we'd have to change the internal helper a bit maybe)

Reply:
Mmm that might help with making run context work, but I'll have to check. From what I saw it is hard to coalesce `Task` and `Task<T>`

Reply:
It is indeed hard and something we have had to do in other places. What we've usually done is made the `Task<T>` the main impl and just use `Task<ValueTuple>` for `Task` equivalents, but it could add an extra layer of stack unnecessarily.